plugin/storage/integration/elasticsearch_test.go

```diff
@@ -105,14 +104,31 @@ func (s *ESStorageIntegration) initializeES(t *testing.T, allTagsAsFields bool)
 		elastic.SetSniff(false))
 	require.NoError(t, err)
 	s.logger, _ = testutils.NewLogger()
-
 	s.client = rawClient
 	s.v8Client, err = elasticsearch8.NewClient(elasticsearch8.Config{
 		Addresses:            []string{queryURL},
 		DiscoverNodesOnStart: false,
 	})
 	require.NoError(t, err)
 
+	// Initialize ES Factory
+	f := es.NewFactory()
+	v, _ := config.Viperize(f.AddFlags)
+	f.InitFromViper(v, zap.NewNop())
+	err = f.Initialize(metrics.NullFactory, s.logger)
```

Comment:
maybe you can use InitFromConfig instead of CLI-based functions. You would need to set some defaults, however.

Reply:
Yes, I was thinking about doing the same thing..
Use NewFactoryWithConfig to initialise instead, will do that!
Thanks a lot

Reply:
Hi, Can you please review and lmk if I am on the right path?